Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -388,43 +393,90 @@ public class ProgramBuilder {
     /// Append a splice from another program.
     public func splice(from program: Program, at index: Int) {
         var idx = index
-        
+
+        // As programs are ref counted, we create copy such that we do not
+        // modify the one in the corpus.
+        let workingCopy = program.copy()
+
+        // The placeholder variable is the next free variable in the victim program.
+        var nextFreeVariableCounter = workingCopy.nextFreeVariable
+        func nextFreeVariable() -> Variable {
+           nextFreeVariableCounter += 1
+           return Variable(number: nextFreeVariableCounter - 1)
+        }
+
+        // Get types of the other program
+        var ai = AbstractInterpreter(for: self.fuzzer)
+
+        for instr in workingCopy {
+            ai.execute(instr)
+        }
+
+        beginAdoption(from: workingCopy)
+
         // Determine all necessary input instructions for the choosen instruction
         // We need special handling for blocks:
         //   If the choosen instruction is a block instruction then copy the whole block
         //   If we need an inner output of a block instruction then only copy the block instructions, not the content
         //   Otherwise copy the whole block including its content
         var needs = Set<Int>()
         var requiredInputs = VariableSet()
-        
+
+        // This maps victim instruction indices to victim : host variable remap
+        // Instead of calling adopt and then using nextvar if the variable is
+        // not in the varMaps map, we do the adoption manually.
+        func rewireOrKeepInputs(instruction: Instruction) {
+            var inputs: [Variable] = Array(instruction.inputs)
+            var needInputs: [Variable] = []
+            for (idx, input) in instruction.inputs.enumerated() {
+                var didRewire = false
+                if probability(0.2) && mode != .conservative {
+                    // TODO(cffsmith): switch to runtime type information when available.
+                    let type = ai.type(of: input) == .unknown ? .anything : ai.type(of: input)
```

Comment:
Nit: maybe write this as
```
var type = ai.type(of: input)
if type == .unknown {
    type = .anything
}
```
(and maybe make a comment that we probably should have a method to do this for us =D )

Reply:
And I guess you can drop the current comment since the runtime type information access refactoring will probably remove `AbstractInterpreter.type(of:)`, and so this code will have to be updated then anyway :)

Reply:
Yeah I had it like that but then I switched to this one-liner, was not sure what the best way is :)

Reply:
Yeah, I find it a bit more readable like this, but it's also just to avoid calling the `type(of:)` method twice (even though that's fine since it currently has no side effects and does a simple lookup) :)